R/explorer.R

```diff
@@ -2,17 +2,40 @@
 #'
 #' This function creates a series of interactive codebook using R htmlwidgets.
 #'
-#' @param dataArray  a list of dataframes (TODO change to tibble?).
-#' @param addEnv  boolean. Indicates whether to add all data frames in current environemnt to explorer.
+#' @param dataList  A list of (optionally named) data frames. Also accepts a character vector of data frame names (must be loaded in environment).
+#' @param addEnv  Logical. Indicates whether to add all data frames in current environemnt to explorer. Defaults to \code{addEnv=TRUE}.
+#' @param demo   Logical. Indicates whether to display demo data frames.  If \code{TRUE}, \code{dataList} and \code{addEnv} settings will be ignored.  Defaults to \code{demo=FALSE}.
 
 #'
 #' @examples
-#' explorer(dataArray=c(cars, iris))
+#' explorer(dataList = list(Cars = mtcars, Iris = iris))
+#' 
+#' explorer(dataList = c("mtcars", "iris"))
 #'
 #' @import htmlwidgets
 #'
 #' @export
-explorer <- function(dataList=c(), addEnv=T, demo=F) {
+explorer <- function(dataList = NULL, addEnv=TRUE, demo=FALSE) {
+
+  # get names of files in dataList as text string
+  if (is.list(dataList)){
+    if(is.null(names(dataList))){
+      names_spec <- rep("", length(dataList))
+    }else{
+      names_spec <- names(dataList)
+    }
+    names_auto <- deparse(substitute(dataList))
+    names_auto <- gsub("list\\(|\\)","", names_auto)
+    names_auto <- unlist(lapply(strsplit(names_auto, split = ","), trimws))
+    
+    out <- data.frame(names_spec, names_auto, stringsAsFactors = FALSE)
+    namesList <- ifelse(out$names_spec=="", out$names_auto, out$names_spec)
+    out <- NULL
+    names(dataList) <- NULL
+  } else { 
+    namesList <- dataList
+  }
+
```

Comment:
I agree with renaming `dataList` to `data`. 

I considered making a new dataList object before, but since I was keeping the part of formatFileList that processes character vectors anyway (for addEnv and demo) I thought I wouldn't bother....we can discuss this...